backend/Testing/FwHeadless/MediaFileTestFixture.cs

```diff
@@ -0,0 +1,168 @@
+using System.Net.Http.Json;
+using LexCore.Entities;
+using Testing.ApiTests;
+using Testing.Services;
+using FwHeadless.Models;
+using Microsoft.AspNetCore.Http.Extensions;
+using System.Net.Http.Headers;
+
+namespace Testing.FwHeadless;
+
+[Trait("Category", "Integration")]
+public class MediaFileTestFixture : ApiTestBase, IAsyncLifetime
+{
+    public Guid ProjectId { get; private set; }
+    public string ProjectCode { get; private set; } = "";
+    public string LoggedInas { get; private set; } = "";
+
+    public async Task LoginIfNeeded(string username)
+    {
+        if (CurrJwt is null)
+        {
+            await LoginAs(username);
+            return;
+        }
+        if (CurrentUser.Username == username) return;
+        await LoginAs(username);
+        return;
+    }
+
+    public async Task InitializeAsync()
+    {
+        await LoginIfNeeded("admin");
+        ProjectCode = Utils.NewProjectCode();
+        ProjectId = await FwHeadlessTestHelpers.CopyProjectToNewProject(HttpClient, ProjectCode, "sena-3");
+        await Utils.AddMemberToProject(ProjectId, this, "manager", ProjectRole.Manager);
+        await Utils.AddMemberToProject(ProjectId, this, "editor", ProjectRole.Editor);
+        // Project folder needs to exist in order for file uploads to be allowed, so trigger first sync including S/R
+        await FwHeadlessTestHelpers.TriggerSync(HttpClient, ProjectId);
+        await FwHeadlessTestHelpers.AwaitSyncFinished(HttpClient, ProjectId);
+    }
+
+    public async Task DisposeAsync()
+    {
+        // Must ensure we're logged in as admin, since some tests might have changed that
+        await LoginIfNeeded("admin");
+        await HttpClient.DeleteAsync($"api/project/{ProjectId}");
+    }
+
+    public async Task<(FileListing?, HttpResponseMessage)> ListFiles(Guid projectId, string? relativePath = null, string loginAs = "admin")
+    {
+        await LoginIfNeeded(loginAs);
+        var url = $"/api/media/list/{projectId}";
+        if (relativePath is not null)
+        {
+            var qb = new QueryBuilder { { "relativePath", relativePath } };
+            url += qb.ToString();
+        }
+        var request = new HttpRequestMessage(HttpMethod.Get, url);
+        var result = await HttpClient.SendAsync(request);
+        return (await result.Content.ReadFromJsonAsync<FileListing>(), result);
+    }
+
+    public async Task<(Guid, HttpResponseMessage)> PostFile(string localPath, string? overrideFilename = null, FileMetadata? metadata = null, string loginAs = "admin", IDictionary<string, string>? extraFields = null)
+    {
+        await LoginIfNeeded(loginAs);
+        var filename = Path.GetFileName(localPath);
+        using (var formData = new MultipartFormDataContent())
+        {
+            if (overrideFilename is not null)
+            {
+                formData.Add(new StringContent(overrideFilename), name: "filename");
```

Comment:
there's no test for this path. There should be a test that this overrides the filename from the file upload.